packages/storage/src/providers/s3/utils/resolveS3ConfigAndInput.ts

```diff
@@ -62,8 +63,16 @@ export const resolveS3ConfigAndInput = async (
 		return credentials;
 	};
 
-	const { bucket, region, dangerouslyConnectToHttpEndpointForTesting } =
+	let { bucket, region, dangerouslyConnectToHttpEndpointForTesting, buckets } =
 		amplify.getConfig()?.Storage?.S3 ?? {};
+
+	const resolvedBucketConfig =
+		apiOptions?.bucket && resolveBucketConfig(apiOptions, buckets);
+
+	if (resolvedBucketConfig) {
+		({ bucket, region } = resolvedBucketConfig);
+	}
+
```

Comment:
instead of mutating the config object, I think it is better to have a common config resolver and move all of this logic inside. 

```suggestion
const { bucket, region, dangerouslyConnectToHttpEndpointForTesting } = resolveStorageConfig(amplify, apiOptions)
```

Reply:
hmm. This is our extracted config resolver right? why abstract it again. I can update the logic to avoid mutating though but in any case i would need to get the config and use that as a fallback.

Reply:
We can always keep breaking down internal implementations to make them reusable and more readable.

Reply:
Agree just dont see a huge difference in doing that here, since its already just adding a single line here.